subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovySourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -156,4 +156,21 @@ class A2{}
         output.contains('Unable to infer source roots. Incremental Groovy compilation requires the source roots and has been disabled. Change the configuration of your sources or disable incremental Groovy compilation.')
         output.contains("Cannot infer source root(s) for source `file '${textFile.absolutePath}'`. Supported types are `File` (directories only), `DirectoryTree` and `SourceDirectorySet`.")
     }
+
+    def 'clear class source mapping file on full recompilation'() {
+        given:
+        source('class A { }')
+        run 'compileGroovy'
+
+        when:
+        file('src/main/groovy/B.java') << 'class B {}'
+        // slightly modify the mapping file
+        file('build/tmp/compileGroovy/source-classes-mapping.txt') << '''org/gradle/MyClass.groovy
```

Comment:
Is it possible to test this without messing with the mappings file?

Reply:
I don't find a cheaper way...